### DISCUSSION TITLE
Firefly show a blank page with Podman + Quadlet

### DISCUSSION THREAD

**Discussion by Froggy232** [firefly-iii/firefly-iii#8909] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hi,
I try to deploy Firefly using podman + quadlet configs file.

So far, I have created three files : 
- `.config/containers/systemd/firefly.network` : https://privatebin.alrhome.net/?8b9e9d9b11ed7da8#Bc6vYtBxKQ3ubNVn1p5paCFzknzpVw6NvLfq3oHiFfk2
- `.config/containers/systemd/firefly-db.container` : https://privatebin.alrhome.net/?b85fde0bac7004d9#45S9am254RYhdRXL9EsTa3B6oMiwFcNp1tVNZW1xH5Ux
- `.config/containers/systemd/firefly-app.container` : https://privatebin.alrhome.net/?21b027d1466a2b4b#4oznDLcZohksDRoqY6xxoEPNPbfGkdbjpgFV3FAaer2Z

When I try to launch them, everything seems to work except I get an `HTTP error 500` when I try to reach it.

The logs from firefly-db are here : https://privatebin.alrhome.net/?3e7f209034dbc870#GviJdipA6Gc2pHGGa7TkJ5XqUMhfwyd9TfvtNZruu3nJ
The logs from firefly-app are here : https://privatebin.alrhome.net/?a3b94b66053443e9#CX7xNrgybPKzZzZ2ZUX189BPg6nribbJwpsvreXWJHUY

Thanks you a lot!

### Debug information

I can't access it, as firefly doesn't launch.

**Comment by JC5** [comment]
I can see the problem: No application encryption key has been specified.

Did you set APP_KEY?

**Comment by Froggy232** [comment]
Hi,
Thanks you a lot for your answer, it was that indeed.
I still have a permission problem, as podman run rootless.
I have tried to add `UserNS=keep-id` to my quadlet file, but it doesn't work.
I will try other things, but would love to get some input if someone have some.
Thanks you!

**Comment by Froggy232** [comment]
Does someone know which UID and GUID is used by firefly container?
Thanks you

**Reply by JC5** [reply]
33, for www-data I think.

**Comment by RequestPrivacy** [comment]
I'm also interested in a working podman & quadlet solution. Unfortunately the pastebin is empty. @Froggy232 maybe you can share your files again?

**Reply by Froggy232** [reply]
Sorry, I have switched from Firefly to ActualBudget, and have removed the config files.
I will try to find them in my backups, and upload them if I still have them.

**Reply by Froggy232** [reply]
I have found them again, I have three files : 

- firefly.network (https://privatebin.alrhome.net/?69bd4d074d70771d#FJexKPQ1Sh42ii939aMpkhpQYHt21effq4dV8CXMDr1s)
- firefly-app.container (https://privatebin.alrhome.net/?1672d902f4d81922#2mMyQf7rtHSoU3XzB95oqsiXk9YjAyzcD33txbAwoB84)
- firefly-db.container (https://privatebin.alrhome.net/?c4b73e88ffec167c#FkyMFYGjtYCTuYm6FtBTZbLA235WqVRRcLNtefFzxm5K)

I think a better approach would be to put them in a pod, but it worked with theses config files few months ago.
Feel free to ask any other questions!

**Reply by RequestPrivacy** [reply]
Thank you for searching in your backups, it is really appreciated. Although really OT do you mind sharing the differences and your subjective pros & cons of ActualBudget vs Firefly as I also try to decide between them.

**Reply by Froggy232** [reply]
Hi again, no problem, happy to help!
ActualBudget is way simpler and easier, with a lot less configurations and options. It's way easier to set it up with podman or docker, and to add accounts link on Nordigen (I think) or GoCardLess, which is my main use-case.
ActualBudget is best suited for beginner or people who don't need advanced options I think, while Firefly III seems way more powerful but also way more complicated.
It's just my opinion after few days of usage, so it's worth what is it.
Feel free to ask any other questions, if you have some!

**Reply by RequestPrivacy** [reply]
Thank you for sharing. I might give Firefly a try and see how the installation goes. Really not a fan of dockers default security so being able to install via podman is my condition for even thinking of hosting such a service myself. Lets see how it goes as my understanding is that you run out of luck trying to install?

Awesome that you might provide further insights, might open a separate discussion and ping you so that this one remains on topic...
